worker/channel_handler.go

```diff
@@ -22,22 +22,26 @@ func handleChannel(channelAction *protos.ChannelAction) {
 	// Apply to all connections for target
 	for _, connection := range connections {
 		if channelAction.Type == protos.ChannelAction_SUBSCRIBE && !common.IncludesString(connection.Channels, channelAction.Channel) {
+			connection.lock.Lock()
 			connection.Channels = append(connection.Channels, channelAction.Channel)
+			connection.lock.Unlock()
```

Comment:
This is the connection (not connections) lock

Reply:
so I think this would be a good way
`
type Channels struct {
	Channels []string
	Mutex    sync.Mutex
}

type SockConnection struct {
	/// WebSocket connection
	Conn    *websocket.Conn
	Id      string
	User    string
	Session string
	/// Message sending channel. Messages sent to it will be sent to the connection
	Sender chan *protos.Message
	/// Channel to close the connect. nil when connection is closed/closing
	CloseChannel chan struct{}
	Channels     Channels
}

func (c *Channels) Add() {

}

func (c *Channels) Remove() {

}
`